SmartCar.ino

```diff
@@ -113,3 +118,59 @@ void stop() {
 void goForward(){
   car.setSpeed(30);
   }
+
+void handleInput(){
+  while (Serial.available()>0){
+    char inChar = Serial.read();
+    switch (inChar) {
+    case 'w': //rotate counter-clockwise
+    setForward();
+    break;
+    case 's': //rotate clock-wise
+      setBackwards();
+      break;
+    case 'a': //go forward
+      setLeft();
+      break;
+    case 'd': //go backward
+      setRight();
+      break;
+    case 'o': //turn back left
+      upSpeed(&speed); 

```

Comment:
No need to overcomplicate things with pointers and references, especially when they are not needed.
For example, if you really want to pass `speed` as an argument, despite it being a global variable and therefore accessible from everywhere, your method signatures should probably look something like this:
```
cpp
void upSpeed(int& speed2) {
 speed2 += 20;
}
```
And then used as `upSpeed(speed);`. But in any case, you should decide whether you need a global variable or not.